### PR TITLE
kustomize: 4.5.4 -> 5.0.2

### PR DESCRIPTION
**Pull request by Djabx**
###### Description of changes

https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]

File: pkgs/development/tools/kustomize/default.nix
Comment (zaninime) [inline pkgs/development/tools/kustomize/default.nix:18]: While we're at it we could change them to the new convention

```suggestion
    hash = "sha256-0pykj3vgnxjaya9krq5l9wvxrfazgcxq0la7dg60wqnbyksl2my1";
```

File: pkgs/development/tools/kustomize/default.nix
Comment (zaninime) [inline pkgs/development/tools/kustomize/default.nix:24]:
```suggestion
  vendorHash = "sha256-wVdB9fTLYg7Ma0dRgDt7X7ncN0+04DyT8kp2+/aQ018=";
```

**Comment by Djabx** [comment]
@zaninime I'm struggling to correct the package outputs. And I'm mearly a user of NixOs, not really a developper of nix (if that make any sense).

Do you have any help I can check to correct the issue here ?

**Comment by Djabx** [comment]
@zaninime I'm using NixOS... but I always feel like I'm doing blakc magic as an sorcerer's apprentice.

Here I just want to update a package version, but how to do it, there is "easy and dump" way to do it. You give insight on `nix hash` (witch I have btw) but how to use it ? on what ?...

So i write `hash = ""` in local... build watch the build crashed telling me the hash it want, replace, rebuild until next step.
I'm not really in power here...

And BTW, that how I removed the vendor line, because with it I had:

```
go: -mod may only be set to readonly when in workspace mode, but it is set to "vendor"
        Remove the -mod flag to use the default readonly value,
        or set GOWORK=off to disable workspace mode.
error: builder for '/nix/store/qk5qsnkzdpmydlq5vpdpy8l4i7yljiax-kustomize-5.0.1.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/pajwdwj8yl6ckf23lpjpc3b7h1bnmr7h-source
       > source root is source
       > patching sources
       > configuring
       > building
       > Building subPackage .
       > go: -mod may only be set to readonly when in workspace mode, but it is set to "vendor"
       >         Remove the -mod flag to use the default readonly value,
       >        or set GOWORK=off to disable workspace mode.
 ```

Removed the vendor line and no more build error...but I feel guilty..

**Comment by zaninime** [comment]
For me without `vendorHash` it does not even start the build.

```
error: either `vendorHash` or `vendorSha256` is required
```

I'll try to have another look later why that error comes up.

**Review by saschagrunert — APPROVED** [review]

**Comment by zaninime** [comment]
@saschagrunert does it build for you?

File: pkgs/development/tools/kustomize/default.nix
Comment (clemak27) [inline pkgs/development/tools/kustomize/default.nix:23]: Hello

I wanted to try this out and encountered the same issues as already mentioned. Not having a `vendorHash` makes the build fail instantly while adding any `vendorHash` leads to the error mentioned in https://github.com/NixOS/nixpkgs/pull/224075#issuecomment-1500060585.

I found this issue https://github.com/NixOS/nixpkgs/issues/203039 that seems to be the reason for this (kustomize uses go workspaces as well)

I set `proxyVendor = true;`, set the `vendorHash` to `""`, replaced it with the actual hash after trying to build it and ended up with this code that works for me:

```suggestion
  proxyVendor = true;
  vendorHash = "sha256-E7DRRf/izgFQhwSKT3NyG+5gwVOd+y3gzC4Q0aWZbVE=";
```


I just don't know if thats an ideal solution :man_shrugging:

Comment (Djabx) [reply pkgs/development/tools/kustomize/default.nix:25]: I've added your modification and no problems on my side.

**Comment by Djabx** [comment]
@zaninime does it build for you ?

**Comment by zaninime** [comment]
No it doesn't. Not in the current state. I was asking @saschagrunert as he approved the PR but I couldn't get it to build it myself.

**Comment by Djabx** [comment]
I've:
* rebase
* update kustomize to v5.0.2
* update all sha.

I've no issue to build it.

@zaninime could you test it please ?

**Review by zaninime — APPROVED** [review]
Now it builds and run on aarch64-darwin.

**Comment by Djabx** [comment]
Could we merge it ?
What are the blockers ?

**Comment by zaninime** [comment]
We need to wait for someone with merge permissions. Now it's approved so it can be merged. Pinging @NickCao as you recently merged a couple other PRs with Go packages.

**Pull request merged**
